app/javascript/controllers/color_avatar_controller.js

```diff
@@ -19,13 +41,23 @@ export default class extends Controller {
     ).toUpperCase();
   };
 
-  handleColorChange(e) {
-    const color = e.currentTarget.value;
+  handleIconChange(e) {
+    const iconSVG = e.currentTarget.closest('label').querySelector('svg').cloneNode(true);
+    this.avatarTarget.innerHTML = '';
+    this.avatarTarget.appendChild(iconSVG);
+  }
+
+  updateAvatarColors(color) {
     this.avatarTarget.style.backgroundColor = `color-mix(in srgb, ${color} 5%, white)`;
     this.avatarTarget.style.borderColor = `color-mix(in srgb, ${color} 10%, white)`;
     this.avatarTarget.style.color = color;
   }
 
+  handleColorChange(e) {
+    const color = e.currentTarget.value;
+    this.updateAvatarColors(color)
+  }
+
   handleParentChange(e) {
```

Comment:
Looks like something happened here with the parent change handling:


https://github.com/user-attachments/assets/263d2342-b67d-439e-a5c0-33330e862c4b